### PR TITLE
OSDOCS-4810: Added module for deleting AWS LZ cluster

### PR DESCRIPTION
**Pull request by dfitzmau**
[OSDOCS-4810](https://issues.redhat.com/browse/OSDOCS-4810)

Version(s):

Issue:
4.12 and 4.13

Link to docs preview: [Deleting a cluster with a configured AWS Local Zone infrastructure](https://56723--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/uninstalling-cluster-aws.html#installation-aws-delete-cluster)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/installation-aws-delete-cluster.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+$ aws cloudformation delete-stack --stack-name <vpn_stack_name>
+----
+
+.Verification
+
+* Check that you removed the stack resources by issuing the following command in the AWS CLI. The AWS CLI outputs that no template components exist. 
++
+[source,terminal]
+----
+$ aws cloudformation describe-stack --stack-name <local_zone_stack_name> <vpn_stack_name>
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:52]: I am not sure if the `--stack-name` receive a list of stacks. The command is also plural: https://docs.aws.amazon.com/cli/latest/reference/cloudformation/describe-stacks.html#options

I think you can add a multi-line here, or just `describe-stacks` without any argument

File: modules/installation-aws-delete-cluster.adoc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+After you install a cluster on Amazon Web Services (AWS) into an existing Virtual Private Cloud (VPC), and you set subnets for each Local Zone location, you can delete the cluster and any AWS resources associated with it. 
+
+The example in the procedure assumes that you created a VPC and its subnets by using a CloudFormation template.
+
+.Prerequisites
+
+* You can access to the copied CloudFormation template and the stack of AWS resources that you created with the template.
+* You can access the installation configuration file that can deploy your cluster.
+* Optional: If you configured SSH access to your cluster, ensure you know the public and private key pair.
+* You logged in to the AWS CloudFormation CLI console.
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:18]: Need to review, for each bullet:

- The user don't need the CloudFormation template, **only the stack name**
- The user don't need the IC, he needs to know the install-dir that the `openshift-install` will point with arg `--dir`
- Dont need SSH in this section
- Need to review that statement. Dont need to access the **AWS Console**, only valid credentials to delete the cloudformation stack. Please take a look to the UPI with CloudFormation as reference: https://docs.openshift.com/container-platform/4.12/installing/installing_aws/installing-aws-user-infra.html#installation-aws-user-infra-delete-bootstrap_installing-aws-user-infra

Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc:18]: Thanks, @mtulio . I was off the mark with these prereqs. I updated the list; please let me know your thoughts.

File: modules/installation-aws-delete-cluster.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+= Deleting a cluster with a configured AWS Local Zone infrastructure
+
+After you install a cluster on Amazon Web Services (AWS) into an existing Virtual Private Cloud (VPC), and you set subnets for each Local Zone location, you can delete the cluster and any AWS resources associated with it. 
+
+The example in the procedure assumes that you created a VPC and its subnets by using a CloudFormation template.
+
+.Prerequisites
+
+* You know the name of the CloudFormation stack, such as `cluster-vpc`. You need the name of the stack to delete the cluster.
+* You can access rights to the directory that contains the installation program. This directory stores any files created by the installation program.
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:16]: -  contains the installation program
- created by the installation program

those sentences become confusing, IMO. The directory contains the files created by the installation program, not necessarily the program need to be there. If the program is there (same dir of install files) you may not need to specify the --dir. I think the first part of this phase must be reviewed.

File: modules/installation-aws-delete-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+.Procedure
+
+. Change to the directory that contains the installation program and delete the cluster by using the `destroy` command:
++
+[source,terminal]
+----
+$ ./openshift-install destroy cluster --dir <installation_directory> \//<1> 
+   --log-level=debug <2>
+----
+<1> For `<installation_directory>`, specify the location of your customized `./install-config.yaml` file.
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:28]: could you please review to be consistent with the phrase above: `This directory stores any files created by the installation program`?
the install-config.yaml will not be there after the cluster is installed (it's processed after the manifests are generated).

File: modules/installation-aws-delete-cluster.adoc (hunk 3 above)
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc:16]: Good point, @mtulio . I updated to state:

> You can access rights to the directory that contains the installation files that were created by the installation program.

File: modules/installation-aws-delete-cluster.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc]: Updated to

> For `<installation_directory>`, specify the location where the installation program stored your installation files.

File: modules/installation-aws-delete-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+.Procedure
+
+. Change to the directory that contains the stored installation files, and delete the cluster by using the `destroy` command:
++
+[source,terminal]
+----
+$ ./openshift-install destroy cluster --dir <installation_directory> \//<1> 
+   --log-level=debug <2>
+----
+<1> For `<installation_directory>`, specify the location where the installation program stored your installation files.
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:28]:
```suggestion
. Change to the directory that contains the installation program, and delete the cluster by using the `destroy cluster` command:
+
[source,terminal]
----
$ ./openshift-install destroy cluster --dir <installation_directory> \//<1> 
   --log-level=debug <2>
----
<1> For `<installation_directory>`, specify the location where the installation files are stored.
```

File: modules/installation-aws-delete-cluster.adoc (hunk 4 above)
Comment (mtulio) [reply modules/installation-aws-delete-cluster.adoc]: the logic is inverted. I suggested the change onto those lines.

**Comment by kalexand-rh** [comment]
I'd rather have destroy presented separately than create. They are rather different use cases.

File: modules/installation-aws-delete-cluster.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc]: How about:

> For `<installation_directory>`, specify the directory that stored any files created by the installation program.

File: modules/installation-aws-delete-cluster.adoc (hunk 5 above)
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc:28]: Updated, but what creates the installation files? The installation program? I would usually need to state what entity does an action.

**Comment by dfitzmau** [comment]
> I'd rather have destroy presented separately than create. They are rather different use cases.

Hi @kalexand-rh and @mtulio . How about placing the deleting a cluster ... procedure in the post-installation doc set? Specifically, https://56723--docspreview.netlify.app/openshift-enterprise/latest/post_installation_configuration/cluster-tasks.html

**Comment by dfitzmau** [comment]
> > I'd rather have destroy presented separately than create. They are rather different use cases.
> 
> (Yeah, my memory was not failing as I thought. =])
> 
> ACK. @dfitzmau maybe we can move that section to [1], mentioning it on the installing page, WDYT?
> 
> [1] https://docs.openshift.com/container-platform/4.12/installing/installing_aws/uninstalling-cluster-aws.html

Thanks, @mtulio . I moved the module to this section.

File: modules/installation-aws-delete-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+[source,terminal]
+----
+$ aws cloudformation delete-stack --stack-name <local_zone_stack_name>
+----
+
+. Delete the stack of resources that represent the VPC:
++
+[source,terminal]
+----
+$ aws cloudformation delete-stack --stack-name <vpn_stack_name>
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:42]:
```suggestion
$ aws cloudformation delete-stack --stack-name <vpc_stack_name>
```

File: modules/cluster-limitations-local-zone.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
```
Comment (mtulio) [inline modules/cluster-limitations-local-zone.adoc:1]: @dfitzmau I think you need to delete this module as it has been moved.

File: modules/installation-aws-delete-cluster.adoc [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+.Verification
+
+* Check that you removed the stack resources by issuing the following commands in the AWS CLI. The AWS CLI outputs that no template component exists. 
++
+[source,terminal]
+----
+$ aws cloudformation describe-stacks --stack-name <local_zone_stack_name> 
+
+$ aws cloudformation describe-stacks --stack-name <vpn_stack_name>
```
Comment (mtulio) [inline modules/installation-aws-delete-cluster.adoc:53]:
```suggestion
$ aws cloudformation describe-stacks --stack-name <vpc_stack_name> 
```

**Review by mtulio — COMMENTED** [review]
@dfitzmau one more review. There are some required changes before final review.

/lgtm cancel

**Review by mtulio — CHANGES_REQUESTED** [review]
@dfitzmau one more review. There are some required changes before final review.

/lgtm cancel

File: modules/installation-aws-delete-cluster.adoc [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+[id="installation-aws-delete-cluster"]
+= Deleting a cluster with a configured AWS Local Zone infrastructure
+
+After you install a cluster on Amazon Web Services (AWS) into an existing Virtual Private Cloud (VPC), and you set subnets for each Local Zone location, you can delete the cluster and any AWS resources associated with it. 
+
+The example in the procedure assumes that you created a VPC and its subnets by using a CloudFormation template.
+
+.Prerequisites
+
+* You know the name of the CloudFormation stack, such as `cluster-vpc`. You need the name of the stack to delete the cluster.
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc:15]: Updated with slight changes.

[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/installation-aws-delete-cluster.adoc:15]: I do not see "Stacks" capitalized in AWS docs. For example: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks.html . I lowercased stacks or this reason.

File: modules/cluster-limitations-local-zone.adoc (hunk 7 above)
Comment (dfitzmau) [reply modules/cluster-limitations-local-zone.adoc]: Hi @mtulio . Well spotted. I removed this module.

**Comment by dfitzmau** [comment]
> @dfitzmau one more review. There are some required changes before final review.
> 
> /lgtm cancel

Thanks, @mtulio . I updated the content based on your feedback.

File: modules/installation-aws-delete-cluster.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+$ aws cloudformation delete-stack --stack-name <vpc_stack_name>
+----
+
+.Verification
+
+* Check that you removed the stack resources by issuing the following commands in the AWS CLI. The AWS CLI outputs that no template component exists. 
++
+[source,terminal]
+----
+$ aws cloudformation describe-stacks --stack-name <local_zone_stack_name>
```
Comment (sheriff-rh) [inline modules/installation-aws-delete-cluster.adoc:51]: One command per [code block](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output) (see the fifth bullet point in this section)

Just make this two code blocks and you're set!

**Review by sheriff-rh — COMMENTED** [review]
One small nit, otherwise great work!

**Pull request merged**